test/src/edu/stanford/nlp/objectbank/LineIteratorTest.java

```diff
@@ -4,32 +4,44 @@
 import java.util.Iterator;
 
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 
 /**
  * @author Christopher Manning
  */
 public class LineIteratorTest {
+    private String s;
+    private String[] output;
+    Iterator<String> di;
 
-  @Test
-  public void testLineIterator() {
-    String s = "\n\n@@123\nthis\nis\na\nsentence\n\n@@124\nThis\nis another\n.\n\n@125\nThis is the\tlast\n";
-    String[] output = { "", "", "@@123", "this", "is", "a", "sentence", "", "@@124", "This", "is another",
-            ".", "", "@125", "This is the\tlast"};
-    Iterator<String> di = new LineIterator<>(new StringReader(s), new IdentityFunction<>());
+    @Before
+    public void setUp() {
+        this.s = "\n\n@@123\nthis\nis\na\nsentence\n\n@@124\nThis\nis another\n.\n\n@125\nThis is the\tlast\n";
+        this.output = new String[]{"", "", "@@123", "this", "is", "a", "sentence", "", "@@124", "This", "is another",
+                ".", "", "@125", "This is the\tlast"};
+        this.di = new LineIterator<>(new StringReader(s), new IdentityFunction<>());
+    }
+
+    @Test
+    public void testWrongLine() {
+        try {
+            for (String out : output) {
+                String ans = di.next();
 
-    try {
-      for (String out : output) {
-        String ans = di.next();
-        // System.out.println(ans);
-        Assert.assertEquals("Wrong line", out, ans);
-      }
-      if (di.hasNext()) {
-        Assert.fail("Too many things in iterator: " + di.next());
-      }
-    } catch (Exception e) {
-      Assert.fail("Probably too few things in iterator: " + e);
+                Assert.assertEquals("Wrong line", out, ans);
+            }
+        } catch (Exception e) {
+            Assert.fail("Probably too few things in iterator: " + e);
+        }
     }
-  }
 
+    @Test
```

Comment:
This is changing the semantics of the test, I believe.  The final `if di.hasNext()` in the original version is checking that the iterator stops when there is no more input.  I don't see that happening here, though.